src/hyperloglog.c

```diff
@@ -465,21 +466,13 @@ int hllPatLen(unsigned char *ele, size_t elesize, long *regp) {
      * Note that the final "1" ending the sequence of zeroes must be
      * included in the count, so if we find "001" the count is 3, and
      * the smallest count possible is no zeroes at all, just a 1 bit
-     * at the first position, that is a count of 1.
-     *
-     * This may sound like inefficient, but actually in the average case
-     * there are high probabilities to find a 1 after a few iterations. */
+     * at the first position, that is a count of 1. */
     hash = MurmurHash64A(ele, elesize, 0xadc83b19ULL);
     index = hash & HLL_P_MASK;      /* Register index. */
     hash >>= HLL_P;                 /* Remove bits used to address the register. */
-    hash |= ((uint64_t)1 << HLL_Q); /* Make sure the loop terminates
-                                       and count will be <= Q+1. */
-    bit = 1;
-    count = 1; /* Initialized to 1 since we count the "00000...1" pattern. */
-    while ((hash & bit) == 0) {
-        count++;
-        bit <<= 1;
-    }
+    hash |= ((uint64_t)1 << HLL_Q); /* Make sure count will be <= Q+1. */
```

Comment:
Since `builtin_ctzll` gurantees will return 0 when count == 0. This would not loop forever or get ub.

Reply:
Not sure I understand this statement. Did you actually mean `hash` instead of `count` in "when count == 0"? We don't pass `count` to `builtin_ctzll`, nor can `count` be 0 in this case.

Reply:
Origin:

```
    hash |= ((uint64_t)1 << HLL_Q); /* Make sure the loop terminates
                                       and count will be <= Q+1. */
```

if not mask like this, the count might be greater than Q+1 since more 0 would be there in `HLL_Q`

Reply:
> Did you actually mean hash instead of count in "when count == 0"? We don't pass count to builtin_ctzll, nor can count be 0 in this case.

Yeah the final 51b can all be 0 here, so we mask the 51B and force it at most to have these

In original code, if `hash(xx) == 0` ( it merely happens), without this mask, it will loop forever, in my patch `builtin_ctzll` handled this case, so I remove the comment for loop